pkg/asset/installconfig/azure/validation.go

```diff
@@ -274,6 +275,7 @@ func ValidatePublicDNS(ic *types.InstallConfig, azureDNS *DNSConfig) error {
 func ValidateForProvisioning(client API, ic *types.InstallConfig) error {
 	allErrs := field.ErrorList{}
 	allErrs = append(allErrs, validateResourceGroup(client, field.NewPath("platform").Child("azure"), ic.Azure)...)
+	allErrs = append(allErrs, validateAzureStackRHCOSImage(ic.Azure.ARMEndpoint, ic.Azure.RHCOSImage, field.NewPath("platform").Child("azure"))...)
```

Comment:
Same as my other comment , we should either only run this conditionally for Azure Stack or return nil for non-azurestack.